src/software/thunderscope/thunderscope_types.py

```diff
@@ -134,9 +134,19 @@ def add_one_widget(self, data: TScopeWidget) -> None:
 
     def refresh(self) -> None:
         """
-        Refreshes all the widgets belonging to this tab
+        Refreshes all the widgets belonging to this tab, and not refresh widget that are not visible.
         """
-        for refresh_func in self.refresh_functions.values():
+        # only refresh dock that are visible
+        if not self.dock_area.isVisible():
+            return
+
+        for widget_name in self.refresh_functions:
+            # only refresh widget inside the dock that are visible
+            self.widgets_map[widget_name]
+            # if not widget.isVisible():
+            #    continue
```

Comment:
Commented out code? We should add it back or remove it.

Reply:
my bad, that was not supposed to be commented.